src/Abp.RedisCache/Runtime/Caching/Redis/AbpRedisCache.cs

```diff
@@ -30,95 +31,255 @@ public class AbpRedisCache : CacheBase
+                {
+                    Logger.ErrorFormat("Unable to set key:{0} to expire at {1:O} asynchronously in Redis", redisKey, DefaultAbsoluteExpireTime.Value.UtcDateTime);
+                }
+            }
+            else
+            {
+                if (!await _database.StringSetAsync(redisKey, redisValue, DefaultSlidingExpireTime))
+                {
+                    Logger.ErrorFormat("Unable to set key:{0} value:{1} to expire after {2:c} asynchronously in Redis", redisKey, redisValue, DefaultSlidingExpireTime);
+                }
+            }
         }
 
-        public override void Set(KeyValuePair<string, object>[] pairs, TimeSpan? slidingExpireTime = null, TimeSpan? absoluteExpireTime = null)
+        public override void Set(KeyValuePair<string, object>[] pairs, TimeSpan? slidingExpireTime = null, DateTimeOffset? absoluteExpireTime = null)
         {
             if (pairs.Any(p => p.Value == null))
             {
                 throw new AbpException("Can not insert null values to the cache!");
             }
 
-            var redisPairs = pairs.Select(p => new KeyValuePair<RedisKey, RedisValue>
-                                          (GetLocalizedRedisKey(p.Key), Serialize(p.Value, GetSerializableType(p.Value)))
-                                         );
+            var redisPairs = pairs.Select(p => {
+                var redisKey = GetLocalizedRedisKey(p.Key);
+                var redisValue = Serialize(p.Value, GetSerializableType(p.Value));
+                return new KeyValuePair<RedisKey, RedisValue>(redisKey, redisValue);
+            });
 
-            if (slidingExpireTime.HasValue || absoluteExpireTime.HasValue)
+            if (!_database.StringSet(redisPairs.ToArray()))
+            {
+                foreach (var pair in redisPairs)
+                {
+                    Logger.ErrorFormat("Unable to multi-set key:{0} value:{1} in Redis", pair.Key, pair.Value);
+                }
+            }
+            else
             {
-                Logger.WarnFormat("{0}/{1} is not supported for Redis bulk insert of key-value pairs", nameof(slidingExpireTime), nameof(absoluteExpireTime));
+                if (absoluteExpireTime.HasValue)
+                {
+                    foreach (var pair in redisPairs)
+                    {
+                        if (!_database.KeyExpire(pair.Key, absoluteExpireTime.Value.UtcDateTime))
+                        {
+                            Logger.ErrorFormat("Unable to multi-set key:{0} to expire at {1:O} in Redis", pair.Key, absoluteExpireTime.Value.UtcDateTime);
+                        }
+                    }
+                }
+                else if (slidingExpireTime.HasValue)
+                {
+                    foreach (var pair in redisPairs)
+                    {
+                        if (!_database.KeyExpire(pair.Key, slidingExpireTime.Value))
```

Comment:
For sliding expire time, let's:
- loop single-key `Set`, instead of
- multi-set + loop single-key expire.